WordPress/WordPressTest/Dashboard/BlogDashboardPostsParserTests.swift

```diff
@@ -0,0 +1,106 @@
+
+        contextManager = TestContextManager()
+        context = contextManager.newDerivedContext()
+        parser = BlogDashboardPostsParser(managedObjectContext: context)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        context = nil
+        contextManager = nil
+    }
+
+    /// When the API return no drafts, but there are local drafts
+    /// Return one local draft
+    func testReturnLocalDraftWhenItExists() {
+        let blog = BlogBuilder(context).build()
+        _ = PostBuilder(context, blog: blog).drafted().build()
+
+        let postsWithLocalContent = parser.parse(cardsResponseWithoutPosts["posts"] as! NSDictionary,
+                                                 for: blog)
+
+        XCTAssertEqual((postsWithLocalContent["draft"] as? [Any])?.count, 1)
+    }
+
+    /// When the API return no drafts, and there are no local drafts
+    /// Return zero drafts
+    func testReturnZeroDraftsWhenNothingLocal() {
+        let blog = BlogBuilder(context).build()
+
+        let postsWithLocalContent = parser.parse(cardsResponseWithoutPosts["posts"] as! NSDictionary,
+                                                 for: blog)
+
+        XCTAssertEqual((postsWithLocalContent["draft"] as? [Any])?.count, 0)
+    }
+
+    /// When the API return drafts, keep the amount returned
+    func testReturnDraftsAsItIsEvenWhenLocalDraftsExists() {
+        let blog = BlogBuilder(context).build()
+        _ = PostBuilder(context, blog: blog).drafted().build()
+
+        let postsWithLocalContent = parser.parse(cardsResponseWithPosts["posts"] as! NSDictionary,
+                                                 for: blog)
+
+        XCTAssertEqual((postsWithLocalContent["draft"] as? [Any])?.count, 3)
+    }
+
+    /// When the API return no scheduled, but there are local scheduled posts
+    /// Return one scheduled post
+    func testReturnLocalScheduledWhenItExists() {
+        let blog = BlogBuilder(context).build()
+        _ = PostBuilder(context, blog: blog).scheduled().withRemote().build()
+
+        let postsWithLocalContent = parser.parse(cardsResponseWithoutPosts["posts"] as! NSDictionary,
+                                                 for: blog)
+
+        XCTAssertEqual((postsWithLocalContent["scheduled"] as? [Any])?.count, 1)
+    }
+
+    /// When the API return no scheduled, and there are no
+    /// local scheduled posts, return zero scheduled posts
+    func testReturnZeroScheduledWhenNothingLocal() {
+        let blog = BlogBuilder(context).build()
+
+        let postsWithLocalContent = parser.parse(cardsResponseWithoutPosts["posts"] as! NSDictionary,
+                                                 for: blog)
+
+        XCTAssertEqual((postsWithLocalContent["scheduled"] as? [Any])?.count, 0)
+    }
+
+    /// When the API return scheduled posts, keep the amount returned
+    func testReturnScheduledAsItIsEvenWhenLocalScheduledExists() {
+        let blog = BlogBuilder(context).build()
+
```

Comment:
I think you're missing a step here of adding a local scheduled post

Reply:
Done in f8d5b80eb64f33f7c0e1d16a1638329a11e70cb3